lighthouse-core/gather/driver.js

```diff
@@ -596,6 +619,26 @@ class Driver {
   }
 
   /**
+   * @return {!Promise<number>}
+   */
+  _getOrCreateIsolatedContextId() {
+    if (typeof this._isolatedExecutionContextId === 'number') {
+      return Promise.resolve(this._isolatedExecutionContextId);
+    }
+
+    return this.sendCommand('Page.getResourceTree')
+      .then(data => {
+        const frameId = data.frameTree.frame.id;
+        return this.sendCommand('Page.createIsolatedWorld', {frameId});
```

Comment:
let's give it a `worldName` while we're at it.

Reply:
done